designs/native-ami-versioning.md

```diff
@@ -0,0 +1,121 @@
+
+### 🔑 Suggestion
+
+Implement Bake Time with `minimumAgeDuration`. Users can take advantage of it with EKS Optimized AMIs generally to follow requirements on AMI bake time. This applies only to AMIs, rather than a Drift TTL which would apply broadly to every provisioning requirement. Users can rely on `minimumAgeDruation` to use it as a safeguard against unexpected AMI upgrades, creating more confidence in relying on Karpenter to make the right decisions.
+
+## Release Pinning
+
+*🔑 Expand amiFamily to allow a pinned version for EKS Optimized AMIs.*
+
+Add an option to specify AMI Release Version with AMI Family. A user can pin known AMI Versions, and declaratively upgrade to a different AMI when ready. This could prevent automatic updates in the event of a newer EKS Optimized AMI. Some users [have requested this](https://github.com/aws/karpenter/issues/1495) with 13 up-votes.
+
+### *Option 1*
+
+Use an AMI Family with an appended magic string. This allows users who only know of version strings to modify AWSNodeTemplates easily to their liking.
+
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  amiFamily: AL2-v20220824 # No version suffix defaults to latest
+```
+
+This benefits current users of EKS Optimized AMIs. It extends cleanly into defaults, and allows users an easy override that follows the same [versioning semantics](#versioning-semantics) used for each of the AMI Families, which users would be aware of.
+
+The downside is users that use custom Bootstrapping logic but do not use EKS Optimized AMIs may find this confusing if they find it in documentation or code. Confusion can be mitigated with good documentation as it’s only referring to EKS Optimized AMIs.
+
+### *Option 2*
+
+Include amiVersion as an additional filter *outside* of the amiSelector.
+
+This may be more intuitive to users that are using EKS Optimized AMIs, but has no value add for those who aren’t. This may be confusing to users looking to extend the amiVersion field into other methods of custom AMI selection, especially since it’s a top level field. One alternative is to make this mutually exclusive with AMISelector so that there’s no room for confusion.
+
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  amiFamily: Bottlerocket
+  amiVersion: v1.9.1 # Defaults to latest, requires AMIFamily if specified
+```
+
+### 🔑 Suggestion
+
+Implement only Option 1. This keeps configuration within the amiFamily field, does not add new fields to be managed, and works well for users who are aware of EKS Optimized semantics, which is the audience for this feature.
```

Comment:
This is easily moved too if we decide to move the amiFamily logic into the amiSelector or some other richer API like in the suggestion [here](https://github.com/aws/karpenter/pull/3204/files#diff-e08ed94660726c8fa676b3fdf3d9c1640b4207419768252f2f19f07825d2fe85).